### PR TITLE
Akka 2.5.20 support

### PR DESCRIPTION
**Pull request by raboof**
Work in progress. Refs #515

### CODE REVIEW

**Comment by johanandren** [comment]
One more API change in ResolvedTarget

File: project/Dependencies.scala [hunk 1]
```diff
@@ -96,6 +97,7 @@ object Dependencies {
     Compile.akkaHttp2Support,
     Compile.akkaDiscovery,
     // these two are available when used through Play, which is also the only case when they are needed
+    Compile.akkaSlf4j % "provided",
```
Comment (ignasi35) [inline project/Dependencies.scala]: The comment above seems to refer to:

```scala
    Compile.play % "provided",
    Compile.playAkkaHttpServer % "provided",
```
The comment should state `these three` or be moved one line down.

File: runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala [hunk 2]
```diff
@@ -74,13 +74,16 @@ object GrpcClientSettings {
...
+    var serviceName = clientConfiguration.getString("service-discovery.service-name")
     val port = clientConfiguration.getInt("port")
     val resolveTimeout = clientConfiguration.getDuration("service-discovery.resolve-timeout").asScala
     val sd = serviceDiscoveryMechanism match {
       case "static" =>
         val host = clientConfiguration.getString("host")
         require(host.nonEmpty, "host can't be empty when service-discovery-mechanism is set to static")
+        // Required by the Discovery infrastructure, even when we use static discovery.
+        if (serviceName.isEmpty)
+          serviceName = "static"
```
Comment (ignasi35) [inline runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala]: Oh wow. Sneaky default is sneaky.

File: project/Dependencies.scala (hunk 1 above)
Comment (renatocaval) [reply project/Dependencies.scala]: moved one line down

**Pull request merged**